stdlib/email/src/main/java/org/ballerinalang/stdlib/email/client/PopClient.java

```
@@ -0,0 +1,125 @@
+            throws BallerinaPopException {
+        log.debug("[PopClient][InitClient] Calling getProperties");
+        Properties properties = PopUtil.getProperties(config, isPop);
+        Session session = Session.getInstance(properties, null);
+        try {
+            Store store;
+
+            if (isPop) {
+                store = session.getStore(PopConstants.POP_PROTOCOL);
+                clientEndpoint.addNativeData(PopConstants.PROPS_HOST,
+                        properties.getProperty(PopConstants.PROPS_POP_HOST));
+            } else {
+                store = session.getStore(ImapConstants.IMAP_PROTOCOL);
+                clientEndpoint.addNativeData(PopConstants.PROPS_HOST,
+                        properties.getProperty(ImapConstants.PROPS_IMAP_HOST));
+            }
+            clientEndpoint.addNativeData(PopConstants.PROPS_STORE, store);
+            clientEndpoint.addNativeData(PopConstants.PROPS_USERNAME,
+                    properties.getProperty(PopConstants.PROPS_USERNAME));
+            clientEndpoint.addNativeData(PopConstants.PROPS_PASSWORD,
+                    properties.getProperty(PopConstants.PROPS_PASSWORD));
+        } catch (NoSuchProviderException e) {
+            log.error("Failed to read message : ", e);
+            throw new BallerinaPopException("Error occurred while accessing POP server", e);
+        }
+    }
+
+    /**
+     * Read emails from the server.
+     * @param clientConnector Represents the POP Client class
+     * @param filter Criteria which are used to read emails
+     * @param isPop True if the protocol is POP3 and false otherwise (if protocol is IMAP)
+     * @return MapValue Returns the type supported to Ballerina
+     * @throws BallerinaPopException If an error occurs in POP client
+     */
+    public static MapValue readMessage(ObjectValue clientConnector, MapValue<Object, Object> filter, boolean isPop)
+            throws BallerinaPopException {
```

IMO, we need to return the error here.

---

Corrected all similar methods with Object return type.